File: Reqnroll/Shared/AppWithRestApi/Services/NotificationService.cs
using WIMP.App.Data;
using WIMP.App.Models;

namespace WIMP.App.Services;

public class NotificationService(IDataRepository repository)
{
    public void SendCancellationNotification(string customerName)
    {
        repository.InsertNotification(new Notification
        {
            CustomerName = customerName,
            Message = "Your order has been cancelled."
        });
    }

    public void SendDelayNotification(string customerName)
    {
        repository.InsertNotification(new Notification
        {
            CustomerName = customerName,
            Message = "Your order has been delayed."
        });
    }

    public IEnumerable<Notification> GetNotifications(string customerName) =>
        repository.GetNotificationsByCustomerName(customerName);
}

File: CucumberJVM/Chapter15/15.7-StubDependencyPattern/Before/src/main/java/com/wimp/app/services/NotificationService.java
package com.wimp.app.services;

import com.wimp.app.data.InMemoryDataStore;
import com.wimp.app.models.Notification;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class NotificationService {
    private final InMemoryDataStore dataStore;

    public NotificationService(InMemoryDataStore dataStore) {
        this.dataStore = dataStore;
    }

    public void sendDelayNotification(String customerName) {
        dataStore.saveNotification(customerName, "Your order is delayed.");
    }

    public List<Notification> getNotifications(String customerName) {
        return dataStore.getNotifications(customerName);
    }
}
